test_bench.py

```diff
@@ -15,41 +15,37 @@
 import pytest
 import time
 import torch
-from bench_utils import workdir, list_model_paths
+from torchbenchmark import workdir, list_models
 
 
 def pytest_generate_tests(metafunc, display_len=24):
     # This is where the list of models to test can be configured
     # e.g. by using info in metafunc.config
-    all_models = list_model_paths()
+    all_models = list(list_models())
     short_names = []
-    for name in all_models:
+    for _, name in all_models:
         short = os.path.split(name)[1]
         if len(short) > display_len:
             short = short[:display_len] + "..."
         short_names.append(short)
-    metafunc.parametrize('model_path', all_models,
+
+    metafunc.parametrize('model', all_models,
                          ids=short_names, scope="class")
     metafunc.parametrize('device', ['cpu', 'cuda'], scope='class')
     metafunc.parametrize('compiler', ['jit', 'eager'], scope='class')
 
 
 @pytest.fixture(scope='class')
-def hub_model(request, model_path, device, compiler):
+def hub_model(request, model, device, compiler):
     """Constructs a model object for pytests to use.
     Any pytest function that consumes a 'modeldef' arg will invoke this
     automatically, and reuse it for each test that takes that combination
     of arguments within the module.
 
     If reusing the module between tests isn't safe, change 'scope' parameter.
     """
-    hubconf_file = 'hubconf.py'
+    Model, model_path = model
     with workdir(model_path):
```

Comment:
will this be needed once your changes to sanitize path manipulation are in?

i guess it's still good to make sure each model runs in a fresh CWD even if we don't gaurantee it runs in its preferred CWD

Reply:
It won't be necessary after the next PR, I will likely change the iterator to no longer return the model_path, and instead add a `benchmark_name` property to the Model object.